### PR TITLE
Implement MD OAuth testing

### PR DESCRIPTION
**Pull request by adreed-msft**
Implements MD OAuth testing by implementing a new resource manager.

This PR will fail testing until Ze returns as we need OAuth access to grab a Managed Disk SAS token and add the OAuth access requirement to one

### CODE REVIEW

File: e2etest/zt_managed_disks_test.go [hunk 1]
```diff
@@ -20,5 +20,56 @@
...
 
+import (
+	"github.com/Azure/azure-storage-azcopy/v10/common"
+	"testing"
+)
+
 // Purpose: Tests for the special cases that relate to moving managed disks (default local VHD to page blob; special handling for
 //     md- and md-impex URLs.
+
+func TestManagedDisks_NoOAuthRequired(t *testing.T) {
```
Comment (vibhansa-msft) [inline e2etest/zt_managed_disks_test.go:31]: Can we have a negative test case where access is revoked and validate transfer attempt after that fails.

Comment (adreed-msft) [reply e2etest/zt_managed_disks_test.go:31]: This feels like it leaves the boundary of testing AzCopy and enters the boundary of testing the service. I don't see the value in adding a test like this.

**Comment by adreed-msft** [comment]
MD OAuth testing is disabled, but this PR should still make it through for the patch release as it contains a fix for one-shot login w/ md oauth

**Review by zezha-msft — APPROVED** [review]
lgtm

**Review by vibhansa-msft — APPROVED** [review]

**Pull request merged**